### PR TITLE
AEA-1274 Rename main_loop to agent_loop in aea/runtime, remove unused assets

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR removes unused assets from docs. It also aligns naming by renaming `main_loop` to `agent_loop`.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/abstract_agent.py [hunk 1]
```diff
@@ -31,6 +31,10 @@ class AbstractAgent(ABC):
     def name(self) -> str:
         """Get agent's name."""
 
+    @abstractproperty
```
Comment (DavidMinarsch) [inline aea/abstract_agent.py:34]: We should maintain the order first properties then methods throughout for public ones. Private ones can go wherever suitable.

File: aea/aea.py [hunk 2]
```diff
@@ -152,7 +152,7 @@ def __init__(
             "Please check connection_ids and resources.connections! No connection left after filtering!",
         )
 
-        self._set_runtime_and_inboxes(
+        self._set_runtime_and_mail_boxes(
```
Comment (DavidMinarsch) [inline aea/aea.py:155]: Improper prior naming

File: aea/aea.py [hunk 3]
```diff
@@ -339,7 +338,7 @@ def handle_envelope(self, envelope: Envelope) -> None:
     def _setup_loggers(self):
         """Set up logger with agent name."""
         for element in [
-            self.runtime.main_loop,
+            self.runtime.agent_loop,
```
Comment (DavidMinarsch) [inline aea/aea.py:341]: Improper prior naming.

File: aea/runtime.py [hunk 4]
```diff
@@ -87,7 +87,31 @@ def __init__(
...
+            loop=loop,
+            threaded=threaded,
+            agent_name=self._agent.name,
+            connections=multiplexer_options["connections"],
+            exception_policy=multiplexer_options.get(
+                "connection_exception_policy", ExceptionPolicyEnum.propagate
+            ),
+            default_routing=multiplexer_options.get("default_routing"),
+            default_connection=multiplexer_options.get("default_connection"),
+            protocols=multiplexer_options.get("protocols", []),
```
Comment (DavidMinarsch) [inline aea/runtime.py:113]: Previously the protocols where missing in the async case. I have no idea why it worked anyway. This is concerning @solarw

File: aea/runtime.py [hunk 5]
```diff
@@ -335,29 +347,23 @@ async def _start_agent_loop(self) -> None:
...
-        return AsyncMultiplexer(
-            connections=multiplexer_options["connections"],
-            threaded=True,
-            agent_name=self._agent.name,
-            exception_policy=multiplexer_options.get(
-                "connection_exception_policy", ExceptionPolicyEnum.propagate
-            ),
-            default_routing=multiplexer_options.get("default_routing"),
-            default_connection=multiplexer_options.get("default_connection"),
-            protocols=multiplexer_options.get("protocols", []),
```
Comment (DavidMinarsch) [inline aea/runtime.py:362]: here it was present. but not above

**Review by solarw — APPROVED** [review]
LGTM

File: aea/runtime.py [hunk 6]
```diff
@@ -117,40 +166,34 @@ def loop(self) -> Optional[AbstractEventLoop]:
...
-        """Create multiplexer instance."""
-        return AsyncMultiplexer(
-            loop=self.loop,
-            agent_name=self._agent.name,
-            connections=multiplexer_options["connections"],
-            exception_policy=multiplexer_options.get(
-                "connection_exception_policy", ExceptionPolicyEnum.propagate
-            ),
-            default_routing=multiplexer_options.get("default_routing"),
-            default_connection=multiplexer_options.get("default_connection"),
```
Comment (DavidMinarsch) [inline aea/runtime.py:135]: Here protocols where missing...

**Pull request merged**